### PR TITLE
more 2.0 progress

### PR DESCRIPTION
**Pull request by sbenedicadb**
- added nullable annotations to Audience.java
- migrated all of AudienceState.java
- progress on AudienceExtension.java

### CODE REVIEW

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 1]
```diff
@@ -211,15 +176,84 @@ void handleEvent(final Event event) {
...
-	// analytics request content
-	void processResponse(final Event event) {
+	// analytics response content
+	void handleAnalyticsResponse(final Event event) {
+		EventData eventData = event.getData();
+
+		if (eventData == null
+				|| !eventData.containsKey(AudienceConstants.EventDataKeys.Analytics.ANALYTICS_SERVER_RESPONSE_KEY)) {
+			Log.warning(LOG_TAG,
+					"hear - Ignoring Analytics response as event data or analytics server response key is unavailable.");
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:186]: will do this in later pr

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 2]
```diff
@@ -211,15 +176,84 @@ void handleEvent(final Event event) {
 		processQueuedEvents();
 	}
 
-	// analytics request content
-	void processResponse(final Event event) {
+	// analytics response content
+	void handleAnalyticsResponse(final Event event) {
+		EventData eventData = event.getData();
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:181]: yes, haven't got here yet :)

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 3]
```diff
@@ -211,15 +176,84 @@ void handleEvent(final Event event) {
...
+		final EventData eventData = event.getData();
+
+		// if we don't have valid data in our event, don't queue it
+		if (eventData == null || !eventData.containsKey(AudienceConstants.EventDataKeys.Lifecycle.LIFECYCLE_CONTEXT_DATA)) {
+			Log.warning(LOG_TAG, "hear - Ignoring Lifecycle response as event data unavailable.");
+			return;
+		}
+
+		Log.trace(LOG_TAG, "hear - queueing the event as we have event data and valid context data.");
+		parentModule.queueAamEvent(event);
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:255]: yup

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 4]
```diff
@@ -211,15 +176,84 @@ void handleEvent(final Event event) {
...
+		}
 	}
 
+	// audience request reset
+	void handleAudienceRequestReset(final Event event) {
+		reset(event);
+	}
+
+	// process shared state change
+	void processSharedStateChange(final Event event) {
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:228]: will be removing

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 5]
```diff
@@ -1190,4 +1224,72 @@ private void clearWaitingEvents() {
...
+	 * visitor {@code profile}, {@code dpid} and {@code dpuuid} for the given {@code pairId}.
+	 *
+	 * @param profile current Audience Manager Visitor Profile {@link Map}
+	 * @param dpid current Data Provider ID
+	 * @param dpuuid current Data Provider User ID
+	 * @param pairId A unique pairing id for one-time listeners
+	 */
+	private void dispatchAudienceResponseIdentity(final Map<String, String> profile, final String dpid, final String dpuuid, final String pairId) {
+		// create the event data from the profile, dpid and dpuuid
+		final EventData eventData = new EventData();
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:1239]: yup, not there yet :)

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 6]
```diff
@@ -1190,4 +1224,72 @@ private void clearWaitingEvents() {
...
+		eventData.putStringMap(AudienceConstants.EventDataKeys.Audience.VISITOR_PROFILE, profile);
+		eventData.putString(AudienceConstants.EventDataKeys.Audience.DPID, dpid);
+		eventData.putString(AudienceConstants.EventDataKeys.Audience.DPUUID, dpuuid);
+		// create the event
+		final Event event = new Event.Builder("Audience Manager Identities", EventType.AUDIENCEMANAGER,
+				EventSource.RESPONSE_IDENTITY).setData(eventData).setPairID(pairId).build();
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:1245]: yup!

File: code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java [hunk 7]
```diff
@@ -17,10 +17,11 @@
...
 
-import static com.adobe.marketing.mobile.audience.AudienceConstants.EX
 import static com.adobe.marketing.mobile.audience.AudienceConstants.EXTENSION_NAME;
 import static com.adobe.marketing.mobile.audience.AudienceConstants.FRIENDLY_EXTENSION_NAME;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.annotation.VisibleForTesting;
 
 import com.adobe.marketing.mobile.*;
```
[the remark this answers is not in the document]
Comment (emdobrin) [reply code/audience/src/main/java/com/adobe/marketing/mobile/audience/AudienceExtension.java:27]: you might accidentally import things from the other extensions using same package, and in particular core (like the EventData class you use right now), so I am thinking we should make explicit imports and use only publicly available APIs.

**Review by emdobrin — APPROVED** [review]
👍 looks good, left the unresolved comments open for next PR(s).

**Pull request merged**